src/platform_impl/macos/view.rs

```diff
@@ -686,6 +768,23 @@ extern "C" fn key_down(this: &Object, _sel: Sel, event: id) {
         };
 
         if pass_along {
+            // Use IME
+            let is_ime_activated: bool = *this.get_ivar("isIMEActivated");
+            if is_ime_activated && is_arrow_or_space_key(virtual_keycode.unwrap()) {
+                (*this).set_ivar("isPreediting", true);
+                let marked_text_ref: &mut id = this.get_mut_ivar("markedText");
+                let composed_string = marked_text_ref.clone().string();
+                let slice = slice::from_raw_parts(
+                    composed_string.UTF8String() as *const c_uchar,
+                    composed_string.len(),
+                );
+                let composed_string = str::from_utf8_unchecked(slice);
+                delete_marked_text(state, composed_string.chars().count() as i32);
+            }
+
+            // Clear them here so that we can know whether they have changed afterwards.
+            let _: () = msg_send![this, clearMarkedText];
+
```

Comment:
I don't understand the comment here. Could you describe this differently? (Maybe this is related to the fact that `interpretKeyEvents` below, may invoke `set_marked_text` or `insert_text`. Regardless it's not clear what this comment means)